src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -560,10 +549,24 @@ private void ProduceTagsSynchronously(TaggerContext<TTag> context)
                     // If we're on the foreground already, we can just update our internal state directly.
                     UpdateStateAndReportChanges(newTagTrees, bufferToChanges, newState, initialTags);
                 }
+                else if (initialTags)
+                {
+                    // If this is the initial set of tags, we fast-track a notification about whatever initial tags we
+                    // computed.  This way the UI is updated quickly for that initial set, and we don't have to wait a
+                    // potentially very long time as the foreground-thread-queue makes it way to our notification.
+                    this.ThreadingContext.JoinableTaskFactory.Run(async () =>
```

Comment:
Why does this need to be in a Run in the first place versus just doing a fire-and-forget? This is going to block up the background thread waiting until the UI thread is able to handle this response, and we don't need that, right?

Reply:
Agreed we should be careful not to block thread pool threads here.

Reply:
Ah, that's a good point.  can address.